restclient/src/main/java/com/box/l10n/mojito/rest/resttemplate/AuthenticatedRestTemplate.java

```diff
@@ -191,7 +195,29 @@ protected URI getURIForResourceAndQueryStringParams(String resourcePath, Map<Str
             uriBuilder.queryParam(entry.getKey(), entry.getValue());
         }
 
-        return uriBuilder.build(false).toUri();
+        // Encode illegal characters in the URI
+        URI uri = uriBuilder.build(false).toUri();
+
+        // Encode DateTime objects in place fully, to ensure the '+' timezone separator doesn't get decoded as a space on the API side
+        uriBuilder = UriComponentsBuilder.fromUri(uri);
+
+        for (Map.Entry<String, ?> entry : queryStringParams.entrySet()) {
+            if (entry.getValue() instanceof DateTime)
```

Comment:
Could this issue be larger than just dates? reading quickly it is not obvious to me at which level we should handle the encoding and for what types.

Reply:
Potentially, but I couldn't say with any certainty.
I noticed a comment in GitBlameWithUsageClient re another area where encoding was mentioned as a potential issue, though not sure what the current status of that is:

```
        // GitBlameWithUsageClient.getGitBlameWithUsages relies on [] not being encoded. keeps this for backward compatibility
        // TODO(spring2) - relaxed-query-chars - should we encode here instead?
        // I'm wondering why spring client doesn't encode those query param, it may be the way we call it that is not
        // good enough since it seems those should be encoded by default
        filterParams.put("repositoryIds[]", repositoryId.toString());
        filterParams.put("offset", offset.toString());
        filterParams.put("limit", batchSize.toString());
```

The current way the client is building the URI ensures _only_ illegal characters are escaped, as there's no .Encode() call in getURIForResourceAndQueryStringParams()
'+' and '[' , ']' are  legal characters - though, depending on the context in which they're used, sometimes they need escaping.
If that call would instead use: uriBuilder.build(false).**encode()**.toUri() , the [] would be escaped to %5B%5D .


In that specific instance though, I'm guessing long term ideally the param names would no longer have [] in their name. Today, there's a couple of collection-type params that have [] in their names:
- repositoryIds[]
- repositoryNames[]
- tmTextUnitIds[]
- localeTags[]

As backward compatibility is likely to be an issue there - one could set up the new API methods with the param names without the [] side by side with the old ones, mark the old ones as deprecated and make sure both methods call the same implementation.

Relevant: 
https://stackoverflow.com/questions/47217647/encoding-query-parameters-with-uricomponentsbuilder
https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/util/UriComponentsBuilder.html#fromHttpUrl-java.lang.String-

Reply:
Side note: I reverted the previous commit in the same branch to preserve the conversation on this PR (and not rewrite the history), assuming the pull request would be merged with the squash option - in order to keep a clean history.